### PR TITLE
247 membership state

### PR DESCRIPTION
**Pull request by motu55**
do not merge before #244 since it contains these commits

### CODE REVIEW

File: juntagrico/dao/memberdao.py [hunk 1]
```diff
@@ -78,33 +79,33 @@ def members_for_create_subscription():
 
     @staticmethod
     def members_for_email():
-        return Member.objects.exclude(inactive=True)
+        return Member.objects.exclude(q_deactivated)
 
     @staticmethod
     def members_for_email_with_subscription():
         return Member.objects.filter(subscriptionmembership__subscription__activation_date__isnull=False).exclude(
```
Comment (pascalfree) [inline juntagrico/dao/memberdao.py:86]: We may want to check here, that the member hasn't left the subscription.

File: juntagrico/entity/member.py [hunk 2]
```diff
@@ -47,18 +47,24 @@ class Member(JuntagricoBaseModel):
...
         _('Kontaktierbar von der Job Seite aus'), default=False)
-
-    canceled = models.BooleanField(_('gekündigt'), default=False)
-    cancelation_date = models.DateField(
+    cancellation_date = models.DateField(
         _('Kündigüngsdatum'), null=True, blank=True)
+    deactivation_date = models.DateField(
+        _('Deaktivierungsdatum'), null=True, blank=True, help_text=_('Sperrt Login und entfernt von E-Mail-Listen'))
     end_date = models.DateField(
         _('Enddatum'), null=True, blank=True)
```
Comment (pascalfree) [inline juntagrico/entity/member.py:55]: add a help text, to clarify what this field means in comparison to the deactivation date.

**Review by pascalfree — CHANGES_REQUESTED** [review]
checking this, I found that the member cancellation form needs to be updated. Will make a separate pull request for that.

File: juntagrico/dao/memberdao.py [hunk 3]
```diff
@@ -78,33 +79,33 @@ def members_for_create_subscription():
...
     @staticmethod
     def members_for_email():
-        return Member.objects.exclude(inactive=True)
+        return Member.objects.exclude(q_deactivated())
 
     @staticmethod
     def members_for_email_with_subscription():
-        return Member.objects.filter(subscriptionmembership__subscription__activation_date__isnull=False).exclude(
-            inactive=True)
+        return Member.objects.filter(MemberDao.has_subscription).exclude(
```
Comment (pascalfree) [inline juntagrico/dao/memberdao.py:86]: `MemberDao.has_subscription` should check that member has not left the given subscription. It should also be aware of future dates. That's actually a bug from another feature.

**Comment by motu55** [comment]
i removed the restrictions since we have the correspondings checks

**Review by pascalfree — APPROVED** [review]

**Pull request merged**